control-plane/p4RuntimeSerializer.cpp

```diff
@@ -1506,7 +1506,47 @@ P4RuntimeSerializer::serializeP4RuntimeIfRequired(const IR::P4Program* program,
                                                   const CompilerOptions& options) {
     // The options parser in the frontend already prints a warning if
     // '--p4runtime-entries-file' is used without '--p4runtime-file'.
-    if (options.p4RuntimeFile.isNullOrEmpty()) return;
+    std::vector<cstring> files;
+    std::vector<P4::P4RuntimeFormat> formats;
+
+    if (!options.p4RuntimeFile.isNullOrEmpty()) {
+        files.push_back(options.p4RuntimeFile);
+        formats.push_back(options.p4RuntimeFormat);
+    }
+    if (!options.p4RuntimeFiles.isNullOrEmpty()) {
+        const char* current = options.p4RuntimeFiles.c_str();
+        while (current != nullptr) {
+            const char* comma = strchr(current, ',');
+            cstring name;
+            if (comma != nullptr) {
+                name = cstring(current, comma - current);
+                current = comma + 1;
+            } else {
+                name = cstring(current);
+                current = nullptr;
+            }
+            files.push_back(name);
+
+            const char* dot = name.find('.');
+            if (dot == nullptr) {
+                formats.push_back(options.p4RuntimeFormat);
+            } else {
+                cstring suffix = cstring(dot);
```

Comment:
The above 5 lines could be more concise as

    if (cstring suffix = name.find('.')) {

If there's no suffix, shouldn't that be an error (like an unrecognized suffix?)